warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -191,28 +201,27 @@ public Value reduce(Key key, Iterator<Value> iter) {
                     // so that we can perform the decrement
                     for (String uid : v.getREMOVEDUIDList()) {
                         
-                        uidsToRemove.add(uid);
-                        
-                        if (uids.contains(uid)) {
-                            
+                        // If we're processing in timestamp order and UIDs contains the removed UID, that means
+                        // a newer key (larger timestamp value) added the UID back in and we don't want to remove
+                        // it here. If we're not processing in timestamp order, then we always want to mark the UID
+                        // as removed and remove it from the UIDs list.
+                        if (!processInTimestampOrder || !uids.contains(uid)) {
```

Comment:
I think I'm of two minds on this. This has repercussions for people using this, and perhaps wonder if this would be better suited in an extension; however, I think I also trust that if you think the comments and option description are sufficient to protect users then I'm good with this.

Reply:
That's a fair point. I'm trying to preserve existing behavior by default (if the option isn't set), but I could add extension points and do my changes that way if you think it's clearer. I'm not sure whether or not either change would be accepted upstream, though.